board/safety/safety_subaru.h

```diff
@@ -218,6 +243,15 @@ static int subaru_tx_hook(CANPacket_t *to_send) {
     violation |= longitudinal_transmission_rpm_checks(transmission_rpm, SUBARU_LONG_LIMITS);
   }
 
+  if (addr == MSG_SUBARU_ES_UDS_Request){
+    int sid = GET_BYTES(to_send, 1, 1);
+
+    // Allow "tester present", "read data by identifier", and "communication control"
+    bool allowed_sid = (sid == 0x3e) || (sid == 0x22) || (sid == 0x28);
+
```

Comment:
why do you need communication control? this can allow programming session, perhaps we should block that

Reply:
Yup. Communication control is required to disable eyesight TX, unless I'm missing something?

Reply:
Ah I suppose we don't check data in ELM mode, so no safety issue here, but init is run before you enter the safety mode, so 0x28 isn't required. (as long as you only need to send communication control once)